server/src/messageHandler.mjs

```diff
@@ -285,7 +280,7 @@ async function handleMessage(message, userId, ws) {
 
   logger.debug(`Sending response for method ${oMsg.method}`);
   let finalResponse = (newResponse ? newResponse : response);
-  if (!process.env.proxy) {
+  if (!process.env.proxy || stateManagement.hasOverride(userId, oMsg.method)) {
```

Comment:
I'm not 100% sure I'm following the code changes here, but doesn't this change make it so that if there's an override, there won't be a response message sent back?!?

Reply:
This was my addition.

I noticed this when testing setting overrides while in proxy mode. We haven't actually done that yet.

This block is saying that if we're not proxying, format a response based on whatever we got from the openRpc/method override. Otherwise if we're in proxy mode just blindly send back the message we got from the endpoint.

If we have an override for a method we need to treat it the same way we would for an openRPC. We need to format the canned response with the typical JsonRPC overhead and then send it.

I find it easier to think of this if/else in reverse: If we're returning a message we got from the thunder proxy endpoint, blindly send it as is. Otherwise we got some kind of mocked message (OpenRPC, method override, etc) that does not have the jsonRpc overhead. Add said overhead and return it.